src/Language/Haskell/Liquid/Bare/OfType.hs

```diff
@@ -166,11 +166,11 @@ ofBRType appRTAlias resolveReft
     goRFun _ x t1 t2 r
       = RFun x <$> go t1 <*> go t2 <*> resolveReft r
 
-    goRApp aliases (RApp tc ts _ r)
+    goRApp aliases !(RApp tc ts _ r)
```

Comment:
This ! and the one below are unnecessary, you're already pattern-matching on the thing.